src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -645,10 +650,35 @@ private void SetOperationStarted()
         {
             if (!_operationStarted)
             {
-                // Since this is the first operation, we set the proxy and server credentials on the
-                // WinRT filter based on the .NET handler's property settings.
+                // Since this is the first operation, we set all the necessary WinRT filter properties.
                 SetFilterProxyCredential();
                 SetFilterServerCredential();
+                if (_serverCertificateCustomValidationCallback != null)
+                {
+                    // The WinRT layer uses a different model for the certificate callback. The callback is
+                    // considered "extra" validation. We need to explicitly ignore errors so that the callback
+                    // will get called.
+                    //
+                    // In addition, the WinRT layer restricts some errors so that they cannot be ignored, such 
+                    // as "Revoked". This will result in behavior differences between UWP and other platforms.
+                    // The following errors cannot be ignored right now in the WinRT layer:
+                    //
+                    //     ChainValidationResult.BasicConstraintsError
+                    //     ChainValidationResult.InvalidCertificateAuthorityPolicy
+                    //     ChainValidationResult.InvalidSignature
+                    //     ChainValidationResult.OtherErrors
+                    //     ChainValidationResult.Revoked
```

Comment:
>Wouldn't it be more future-proof to just add ALL ChainValidationResult values to the IgnorableServerCertificateErrors, and just comment that some of them will not effectively be ignored?

Yes, but passing in ones that can't be ignored throws exceptions from the Add() method.  And I didn't want to add exception throwing noise all the time even if we catch it. It adds noise to debugging sessions.